pkg/admission/conf/am_conf.go

```diff
@@ -118,7 +119,11 @@ func NewAdmissionControllerConf(configMaps []*v1.ConfigMap) *AdmissionController
 }
 
 func (acc *AdmissionControllerConf) RegisterHandlers(configMaps informersv1.ConfigMapInformer) {
-	configMaps.Informer().AddEventHandler(&configMapUpdateHandler{conf: acc})
+	_, err := configMaps.Informer().AddEventHandler(&configMapUpdateHandler{conf: acc})
+	if err != nil {
+		log.Log(log.AdmissionConf).Fatal("Failed to create Register handler", zap.Error(err))
+		os.Exit(1)
```

Comment:
Maybe we can return error instead of breaking program here, since the caller may want to do something cleanup before leaving.

Reply:
Update to return error for NewNamespaceCache and RegisterHandler functions back to main(), and it will stop the process. Check the new commit.
Another question: what about the cleanup here?

Reply:
> Another question: what about the cleanup here?

It seems to me propagating the error to callers allow them to do more actions. For example, log with more details or sent event or release resources. At any rate, exiting program in the middle of call chain is anti-pattern to me.

Reply:
Good insight. Many thanks. Allow NewNamespaceCache, RegisterHandlers functions to pass error back, and deal with the error in main(). The code is updated.